tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -441,23 +410,23 @@ export class WebGPUBackend extends KernelBackend {
 
     // Download the values from the GPU.
     let vals: backend_util.BackendValues;
-    if (info.dtype === 'complex64') {
+    if (tensorData.dtype === 'complex64') {
       const ps = await Promise.all([
-        this.read(info.complexTensorInfos.real.dataId),
-        this.read(info.complexTensorInfos.imag.dataId)
+        this.read(tensorData.complexTensorInfos.real.dataId),
+        this.read(tensorData.complexTensorInfos.imag.dataId)
       ]);
 
       const realValues = ps[0];
       const imagValues = ps[1];
       vals = backend_util.mergeRealAndImagArrays(
           realValues as Float32Array, imagValues as Float32Array);
     } else {
-      const data = info.values != null ?
-          info.values :
-          await this.getBufferData(
-              info.bufferInfo.buffer, info.bufferInfo.size);
-      vals =
-          webgpu_util.ArrayBufferToTypedArray(data as ArrayBuffer, info.dtype);
+      const bufferInfo = tensorData.resourceInfo as BufferInfo;
+      const data = tensorData.values != null ?
```

Comment:
nit: `tensorData.values` has been checked in L403. So here we can always use `const data = await this.getBufferData`?